lighthouse-cli/test/smokehouse/smokehouse-report.js

```diff
@@ -102,34 +102,64 @@ function findDifference(path, actual, expected) {
   return null;
 }
 
+/**
+ * @param {string} name
+ * @param {any} actualResult
+ * @param {any} expectedResult
+ * @returns {Smokehouse.Assertion}
+ */
+function makeAssertion(name, actualResult, expectedResult) {
+  const diff = findDifference(name, actualResult, expectedResult);
+
+  return {
+    category: name,
+    actual: actualResult,
+    expected: expectedResult,
+    equal: !diff,
+    diff,
+  };
+}
+
 /**
  * Collate results into comparisons of actual and expected scores on each audit.
- * @param {Smokehouse.ExpectedLHR} actual
- * @param {Smokehouse.ExpectedLHR} expected
- * @return {Smokehouse.LHRComparison}
+ * @param {Smokehouse.ExpectedRunResult} actual
+ * @param {Smokehouse.ExpectedRunResult} expected
+ * @return {Smokehouse.RunComparison}
  */
 function collateResults(actual, expected) {
-  const auditNames = Object.keys(expected.audits);
-  const collatedAudits = auditNames.map(auditName => {
-    const actualResult = actual.audits[auditName];
-    if (!actualResult) {
-      throw new Error(`Config did not trigger run of expected audit ${auditName}`);
-    }
+  /** @type {Smokehouse.Assertion[]} */
+  let artifactAssertions = [];
+  if (expected.artifacts) {
+    const artifactNames = /** @type {(keyof LH.Artifacts)[]} */ (Object.keys(expected.artifacts));
+    artifactAssertions = artifactNames.map(artifactName => {
+      const actualResult = (actual.artifacts || {})[artifactName];
+      if (!actualResult) {
+        throw new Error(`Config run did not generate artifact ${artifactName}`);
+      }
 
-    const expectedResult = expected.audits[auditName];
-    const diff = findDifference(auditName, actualResult, expectedResult);
+      const expectedResult = (expected.artifacts || {})[artifactName];
+      return makeAssertion(artifactName + ' artifact', actualResult, expectedResult);
+    });
+  }
 
-    return {
-      category: auditName,
-      actual: actualResult,
-      expected: expectedResult,
-      equal: !diff,
-      diff,
-    };
-  });
+  /** @type {Smokehouse.Assertion[]} */
+  let auditAssertions = [];
+  if (expected.lhr.audits) {
+    auditAssertions = Object.keys(expected.lhr.audits).map(auditName => {
+      const actualResult = actual.lhr.audits[auditName];
+      if (!actualResult) {
+        throw new Error(`Config did not trigger run of expected audit ${auditName}`);
+      }
+
+      const expectedResult = expected.lhr.audits[auditName];
+      return makeAssertion(auditName + ' audit', actualResult, expectedResult);
+    });
+  }
+
+  const assertions = [...artifactAssertions, ...auditAssertions];
 
   return {
-    audits: collatedAudits,
+    assertions,
```

Comment:
a bit strange that we separate out the others but mix these into `assertions` IMO, I might prefer a separate artifacts property or just collapsing everything into a comparison array

but not a huge deal

Reply:
Merged to just return a Comparison array.